### PR TITLE
[Enhancement](s3-load) Add domain connection and aksk correction check for S3 load

### PR DESCRIPTION
**Pull request by liaoxin01**
## Proposed changes

Add domain connection and aksk correction check for S3 load before actual execution.

### CODE REVIEW

**Comment by liaoxin01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java [hunk 1]
```diff
@@ -517,6 +522,25 @@ public void analyze(Analyzer analyzer) throws UserException {
...
+    private String getProviderFromEndpoint(String endpoint) {
+        for (String provider : PROVIDERS) {
+            if (endpoint.toLowerCase().contains(provider.toLowerCase())) {
+                return provider;
+            }
+        }
+        return "s3";
+    }
+
+    private String getBucketFromFilePath(String filePath) throws Exception {
```
[the remark this answers is not in the document]
Comment (gavinchou) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/LoadStmt.java:535]: s3://bucket/prefix/obj

s3://bucket/prefix/obj*/**/a

https://{accountName}.blob.core.windows.net/{containerName}/blobName

-- azure
load with s3 {
ak = {accountName} 
sk = {sk}
endpoint = {accountName}.blob.core.windows.net -- optional
bucket = {containerName}
}

**Comment by liaoxin01** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**